libmamba/src/solver/libsolv/helpers.cpp

```diff
@@ -347,14 +392,70 @@ namespace mamba::solver::libsolv
                 }
             }
         }
+
+        void set_repo_solvables_with_sigs(
+            solv::ObjPool& pool,
+            solv::ObjRepoView repo,
+            const specs::CondaURL& parsed_url,
+            const std::string& channel_id,
+            const std::string& default_subdir,
+            const simdjson::dom::object& packages,
+            const simdjson::dom::object& repodata,
+            bool verify_artifacts
+        )
+        {
+            if (auto signatures = repodata["signatures"].get_object();
+                !signatures.error() && verify_artifacts)
+            {
+                set_repo_solvables(
+                    pool,
+                    repo,
+                    parsed_url,
+                    channel_id,
+                    default_subdir,
+                    packages,
+                    signatures.value()
+                );
+            }
+            else
+            {
+                // NOTE We need to create a fake signatures json doc to get a valid
+                // simdjson::dom::object (otherwise we get a segfault because constructor yields to
+                // an invalid simdjson::dom::object)
+                simdjson::dom::parser fake_parser;
+                simdjson::dom::object fake_signatures;
+                get_fake_signatures(fake_parser, fake_signatures);
+                set_repo_solvables(
+                    pool,
+                    repo,
+                    parsed_url,
+                    channel_id,
+                    default_subdir,
+                    packages,
+                    fake_signatures
+                );
```

Comment:
What if we use a check (`if` or `constexpr if`) around the setting of the signature in `set_solvable`, would that simplify this fake parser?

Reply:
Well we are actually checking if the signatures are empty around that indeed by getting the object in [here](https://github.com/mamba-org/mamba/pull/3192/files#diff-e2b0f07cf93b08e4834c9512510e360569da2cddcd70c333eea783269510c173R169). And there is no other way (at least that I could find) to check that with simdjson objects unless we use this fake signatures object workaround.

Reply:
What if passing down a `std::optional<simdjson::dom::object>` as signature to `set_repo_solvables`, would that avoid creating the fake parser?

Reply:
Good point!